File: LektaAndroid/app/src/main/java/com/lekta/app/models/Operador.kt
package com.lekta.app.models

import java.security.MessageDigest
import java.util.UUID

data class Operador(
    val id: String = UUID.randomUUID().toString(),
    val name: String,
    val pinHash: String,
    var isActive: Boolean = true,
    val createdAt: Long = System.currentTimeMillis()
) {
    fun verifyPIN(pin: String): Boolean = pinHash == hashPIN(pin)

    companion object {
        fun hashPIN(pin: String): String {
            val digest = MessageDigest.getInstance("SHA-256")
            return digest.digest(pin.toByteArray(Charsets.UTF_8))
                .joinToString("") { "%02x".format(it) }
        }
    }
}

File: WarehouseApp/Models/Operador.swift
import Foundation
import CryptoKit

struct Operador: Identifiable, Codable {
    var id: UUID = UUID()
    var name: String
    var pinHash: String
    var isActive: Bool = true
    var createdAt: Date = Date()

    static func hashPIN(_ pin: String) -> String {
        let digest = SHA256.hash(data: Data(pin.utf8))
        return digest.map { String(format: "%02x", $0) }.joined()
    }

    func verifyPIN(_ pin: String) -> Bool {
        pinHash == Operador.hashPIN(pin)
    }
}
